packages/core-transaction-pool/src/memory.ts

```diff
@@ -274,7 +274,7 @@ export class Memory {
     }
 
     private currentHeight(): number {
-        if (app.has("state")) {
+        if (app && typeof app.has === "function" && app.has("state")) {
```

Comment:
What's the purpose of this? If you are getting an error here during tests it means that a mock is missing and not that you should add more checks to get a test passing.

Reply:
Removed